### PR TITLE
Add table of contents to help page

### LINKED ISSUE #998: Add table of contents for Help page
**Description**:
We had had some feedback that it would be convenient to add a Table of contents (at the top of) the Help page. This would be certainly useful for the users. Could we look at automating this by checking the existing headers?

**Acceptance criteria**:
- [ ] Add table of contents in the Help page, ideally in an automated way

### PR DESCRIPTION
**Pull request by louise-davies**
## Description
Add a TOC that is automatically generated based on finding headings (i.e. `h1`, `h2` etc) with IDs (they need IDs so we can create anchor links to them).

I've also refactored the help page to just have one "section" that the admin can configure, as this is easier from an admin perspective (can edit the document as a whole rather than distinct sections, as well as it being more flexible) as well as from a code perspective (fewer LOC in help page!). Also, this will be needed in the future with OperationsGateway anyway!

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] Paste in the current DLS help text and see that it adds nested TOC links for scarf and globus help

## Agile board tracking
Closes #998

### CODE REVIEW

File: src/helpPage/helpPage.component.tsx [hunk 1]
```diff
@@ -53,81 +56,71 @@ interface HelpPageProps {
...
+    else if (currLevel < hLevel) {
+      tocHtml += '<ul>'.repeat(hLevel - currLevel);
+    }
+    // we are at the same level, so just add ourselves
+    tocHtml += `<li style='list-style: none'><a href='#${h.id}'>${h.textContent}</a></li>`;
+    currLevel = hLevel;
+  }
+  // close off any remaining uls
+  tocHtml += '</ul>'.repeat(currLevel - 2);
+  console.log(tocHtml);
```
Comment (sam-glendenning) [inline src/helpPage/helpPage.component.tsx:92]: Remove this line

**Review by sam-glendenning — CHANGES_REQUESTED** [review]

File: public/res/default.json [hunk 2]
```diff
@@ -68,22 +68,12 @@
...
...ill submit your download request to be processed. You should see a <strong>Cart successfully submitted</strong> notification popup on the right corner. The request will be added inside the downloads window.<h2 id='download-title'>Download</h2>Clicking on the above will open up the downloads window:<br><img src='/res/images/downloads-button.png'><br>The Downloads window is a grid that list all your download requests.<br><img src='/res/images/downloads.png'><br>The <strong>Status</strong> column gives you the current status of your download. Possible values are <strong>Restoring from Tape</strong> and <strong>Available</strong>. The <strong>Download</strong> button is disabled when the status is restoring. Once available, you can click the download button to download the file via the browser if <strong>Https</strong> transport type was used for that request.<br><br>Please note that download pausing and resuming is not supported. Multi-part download managers are also not supported.<br><br>For downloads using Globus as the transport type, when the status is complete, you can login to <a href='https://www.globus.org/SignIn'>Globus Online</a> and initiate file transfer to your endpoint."
```
[the remark this answers is not in the document]
Comment (louise-davies) [reply public/res/default.json:76]: Editing the HTML in the JSON is hard anyway, I normally copy it out and edit it as an HTML file, which made splitting it up and putting it in 6 fields worse 😅 Some of the typos will be from the original TopCAT help page, but some are obviously copy/paste errors. This is also just a default value which gets overridden anyway but I'll adjust it

File: src/helpPage/helpPage.component.test.tsx [hunk 3]
```diff
@@ -37,4 +40,13 @@ describe('Help page component', () => {
     );
     expect(wrapper).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (louise-davies) [reply src/helpPage/helpPage.component.test.tsx:41]: I did try using shallow for this, but the problem is that enzyme doesn't work nicely with the `MuiThemeProvider` and you can't `dive` past it. I can just test `HelpPageWithoutStyles` - it just reduced the coverage is all iirc. I also didn't want to make too many changes as I expect this was fixed in the MUIv5 PR

[the remark this answers is not in the document]
Comment (louise-davies) [reply src/helpPage/helpPage.component.test.tsx:41]: I'll leave it alone then, as changing this test will only cause more merge conflicts in the MUIv5 branch and the issue is fixed in that branch

**Review by sam-glendenning — APPROVED** [review]
As per my last commit, made some minor changes to the header element IDs by removing the -title suffix. This allows for a more user-friendly URL, e.g. /help#browse compared to /help#browse-title

Otherwise, looks good to me!

**Pull request merged**